### PR TITLE
[UWP] Fixes for usage of XF with .net native toolchain

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change ###

Adding the xaml assembly so reflection works.
Add reference on rd.xml to use Windows.UI.Xaml.

### Bugs Fixed ###

https://bugzilla.xamarin.com/show_bug.cgi?id=56795

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Rebased on top of master at time of PR
- [ ] Changes adhere to coding standard
- [ ] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Platform.WinRT/WindowsBasePlatformServices.cs [hunk 1]
```diff
@@ -74,10 +74,14 @@ public virtual Assembly[] GetAssemblies()
 			}
 
 			Assembly thisAssembly = GetType().GetTypeInfo().Assembly;
+			Assembly xamlAssembly = typeof(Xamarin.Forms.Xaml.IMarkupExtension).GetTypeInfo().Assembly;
```
Comment (samhouts) [inline Xamarin.Forms.Platform.WinRT/WindowsBasePlatformServices.cs]: Not a big deal, but I feel like this should be down next to line 82.

File: Xamarin.Forms.Platform.UAP/Properties/Xamarin.Forms.Platform.UAP.rd.xml [hunk 2]
```diff
@@ -28,6 +28,7 @@
   <Library Name="Xamarin.Forms.Platform.UAP">
 
   	<!-- add directives for your library here -->
-
+    <Namespace Name="Windows.UI.Xaml" MarshalObject="Required All"/>
```
Comment (samhouts) [inline Xamarin.Forms.Platform.UAP/Properties/Xamarin.Forms.Platform.UAP.rd.xml]: We can actually just do `<Namespace Name="Windows.UI.Xaml.Automation.Peers" MarshalObject="Required All"/> `. I think bringing in the whole namespace is bloaty.

Comment (samhouts) [inline Xamarin.Forms.Platform.UAP/Properties/Xamarin.Forms.Platform.UAP.rd.xml]: But if it works, it works. ;)

Comment (rmarinho) [inline Xamarin.Forms.Platform.UAP/Properties/Xamarin.Forms.Platform.UAP.rd.xml]: Oh nice. let me try that ..

File: Xamarin.Forms.Platform.WinRT/WindowsBasePlatformServices.cs [hunk 3]
```diff
@@ -78,6 +78,10 @@ public virtual Assembly[] GetAssemblies()
 			if (!assemblies.Contains(thisAssembly))
 				assemblies.Add(thisAssembly);
 
+			Assembly xamlAssembly = typeof(Xamarin.Forms.Xaml.IMarkupExtension).GetTypeInfo().Assembly;
+			if (!assemblies.Contains(xamlAssembly))
+				assemblies.Add(xamlAssembly);
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.WinRT/WindowsBasePlatformServices.cs]: did we already had a dependency on that assembly ? or are we adding a new one ?

also, using `typeof(XamlLoader)` means more than using `IMarkupExtension` and helps understand which assembly we are talking about

**Pull request merged**